innerproductargument/innerproductargument_test.go

```diff
@@ -1,98 +1,150 @@
 package innerproductargument
 
 import (
-	"bytes"
+	"fmt"
 	"testing"
+	"time"
 
 	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
 	"github.com/consensys/gnark-crypto/ecc/bls12-381/fr"
 	"github.com/jsign/curdleproofs/common"
-	"github.com/jsign/curdleproofs/msmaccumulator"
+	"github.com/jsign/curdleproofs/group"
 	"github.com/jsign/curdleproofs/transcript"
 	"github.com/stretchr/testify/require"
 )
 
+type testConfig struct {
+	name                  string
+	group                 group.Group
+	genRandomGroupElement func(*common.Rand) (group.Element, error)
+}
+
 func TestInnerProductArgument(t *testing.T) {
 	t.Parallel()
 
 	n := 128
-	var proof Proof
-	{
-		transcript := transcript.New([]byte("IPA"))
-		crs, B, C, z, bs, cs, _ := setup(t, n)
 
-		rand, err := common.NewRand(42)
-		require.NoError(t, err)
-		proof, err = Prove(
-			crs,
-			B,
-			C,
-			z,
-			bs,
-			cs,
-			transcript,
-			rand,
-		)
-		require.NoError(t, err)
+	configs := []testConfig{
+		{
+			name:  "G1",
+			group: &group.GroupG1{},
+			genRandomGroupElement: func(rand *common.Rand) (group.Element, error) {
+				randG1Aff, err := rand.GetG1Affine()
+				if err != nil {
+					return nil, err
+				}
+				var randG1Jac bls12381.G1Jac
+				randG1Jac.FromAffine(&randG1Aff)
+				return group.FromG1Jac(randG1Jac), nil
+			},
+		},
+		{
+			name:  "Gt",
+			group: &group.GroupGt{},
+			genRandomGroupElement: func(rand *common.Rand) (group.Element, error) {
+				randGt, err := rand.GetGt()
+				if err != nil {
+					return nil, err
+				}
+				return group.FromGt(randGt), nil
+			},
+		},
+	}
+	for _, config := range configs {
+		t.Run(config.name, func(t *testing.T) {
+
+			var proof Proof
+			{
+				transcript := transcript.New([]byte("IPA"))
+				crs, B, C, z, bs, cs, _ := setup(t, config, n)
+
+				rand, err := common.NewRand(42)
+				require.NoError(t, err)
+				start := time.Now()
+				proof, err = Prove(
+					config.group,
+					crs,
+					B,
+					C,
+					z,
+					bs,
+					cs,
+					transcript,
+					rand,
+				)
+				require.NoError(t, err)
+				fmt.Printf("Prove for %s took %s\n", config.name, time.Since(start))
+			}
+
+			t.Run("completeness", func(t *testing.T) {
+				transcript := transcript.New([]byte("IPA"))
+				msmAccumulator := group.NewMsmAccumulator(config.group)
+				crs, B, C, z, _, _, us := setup(t, config, n)
+
+				rand, err := common.NewRand(43)
+				require.NoError(t, err)
+
+				startVerify := time.Now()
+				ok, err := Verify(
+					config.group,
+					proof,
+					crs,
+					B,
+					C,
+					z,
+					us,
+					transcript,
+					msmAccumulator,
+					rand,
+				)
+				require.NoError(t, err)
+				require.True(t, ok)
+
+				startMsmAccumCheck := time.Now()
+				ok, err = msmAccumulator.Verify()
+				require.NoError(t, err)
+				require.True(t, ok)
+				fmt.Printf("Verify for %s took %s (%s+%s)\n", config.name, time.Since(startVerify), startMsmAccumCheck.Sub(startVerify), time.Since(startMsmAccumCheck))
```

Comment:
I thought can be interesting to show time with a bit more detail here.
As mentioned in the PR description:
`Verify for Gt took 57.649203ms (12.271846ms+45.377507ms)`

This means a total of 57ms, where 12ms is the verification phase, and 45ms to verify the MSM accumulator. 

This shows that most of the time is verifying the accumulated MSM, which prob makes sense considering the mentioned naive impl that we have today.